File: qcblog-service/src/main/resources/static/js/service/userService.js
app.service('userService',function($http){
    this.findUserAll = function () {
        return $http.get("/User/findAll");
    };
    //增加
    this.add=function(entity){
        return  $http.post('/User/add',entity );
    };
    //计数
    this.count=function(){
        return  $http.get('/User/count');
    };
    //修改
    this.update=function(entity){
        return  $http.post('/User/update',entity);
    };
    //批量删除
    this.dele=function(ids){
        return $http.get('/User/delete?ids='+ids);
    };
    //单体删除
    this.deleOne=function(id){
        return $http.get('/User/deleteOne?id='+id);
    };
    //禁用
    this.forbidden=function (name) {
      return $http.get("/User/forbiddenOne?name="+name);
    };
    //启用
    this.enabled=function (name) {
        return $http.get("/User/enabledOne?name="+name);
    };
    this.search=function(page,rows,searchEntity){
        return $http.post('/User/search?page='+page+"&rows="+rows, searchEntity);
    };
    //分页
    this.findPage=function(page,rows){
        return $http.get('/User/findPage?page='+page+'&rows='+rows);
    };
    //名称查询
    this.findUserOne=function(name){
        return $http.get('/User/findOne?name='+name);
    }
});

File: qcblog-service/src/main/java/com/qcblog/service/UserService.java
package com.qcblog.service;

import com.github.pagehelper.Page;
import com.github.pagehelper.PageHelper;
import com.qcblog.common.PageResult;
import com.qcblog.mapper.UserMapper;
import com.qcblog.pojo.User;
import com.qcblog.pojo.UserExample;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserMapper userMapper;


    public User findUserByName(String name) {
        User user = new User();
        user.setUsername(name);
        return userMapper.selectOne(user);
    }
    public void insert(User user){
        userMapper.insert(user);
    }

    public List<User> findAll() {
        return userMapper.selectByExample(null);
    }

    public void add(User user) {
        userMapper.insert(user);
    }


    public User findOne(String name) {
        return userMapper.findByName(name);
    }
    /**
     * 修改
     */
    public void update(User user){
        userMapper.update(user);
    }

    /**
     * 禁用启用
     */
    public void updateForbidden(User user){
        userMapper.updateForbidden(user);
    }
    /**
     * 批量删除
     * @param ids
     */
    public void delete(Integer[] ids) {
        for(Integer id:ids){
            userMapper.deleteByPrimaryKey(id);
        }
    }
    /**
     * 单体删除
     * @param user
     */
    public void deleteOne(User user) {
        userMapper.deleteOne(user);
    }

    /**
     * 通过ID 查询
     * @return
     */
    public User selectById(Integer id){
        return userMapper.selectByPrimaryKey(id);
    }

    public String countById(){
        return userMapper.countById();
    }
    public PageResult findPage(User user, int pageNum, int pageSize) {
        PageHelper.startPage(pageNum, pageSize);
        UserExample example=new UserExample();
        UserExample.Criteria criteria = example.createCriteria();
        if(user!=null){
            if(user.getUsername()!=null && user.getUsername().length()>0){
                criteria.andUsernameLike("%"+user.getUsername()+"%");
            }
            if(user.getPassword()!=null && user.getPassword().length()>0){
                criteria.andPasswordLike("%"+user.getPassword()+"%");
            }

            if(user.getStatus()!=null && user.getStatus().length()>0){
                criteria.andStatusEqualTo(user.getStatus());
            }
        }
        Page<User> page= (Page<User>)userMapper.selectByExample(example);
        return new PageResult(page.getTotal(), page.getResult());
    }

    /**
     * 按分页查询
     */

    public PageResult findPage(int pageNum, int pageSize) {
        PageHelper.startPage(pageNum, pageSize);
        Page<User> page=   (Page<User>) userMapper.selectByExample(null);
        return new PageResult(page.getTotal(), page.getResult());
    }
    public int countByArticleUserId(){
        return userMapper.countByArticleUserId();
    }
    public void updateCarticnum(Integer carticnum){
        userMapper.updateCarticnum(carticnum);
    }
}
